src/app/app.module.ts

```diff
@@ -1,30 +1,94 @@
-import { SharedModule } from './shared/shared.module';
 import { BrowserModule } from '@angular/platform-browser';
 import { NgModule } from '@angular/core';
+import { ReactiveFormsModule } from '@angular/forms';
+import { HttpModule } from '@angular/http';
 import { RouterModule } from '@angular/router';
+import {
+  MdToolbarModule,
+  MdButtonModule,
+  MdCheckboxModule,
+  MdSidenavModule,
+  MdSelectModule,
+  MdIconModule,
+  MdIconRegistry,
+  MdInputModule,
+  MdListModule,
+  MdOptionModule,
+  MdProgressSpinnerModule,
+  MdCardModule
+} from '@angular/material';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
 import { StoreModule } from '@ngrx/store';
+import { EffectsModule } from '@ngrx/effects';
+import { StoreRouterConnectingModule } from '@ngrx/router-store';
 
-import { routes } from './app.routes';
 import { AppComponent } from './app.component';
 import { HeaderComponent } from './components/header/header.component';
 import { SidenavComponent } from './components/sidenav/sidenav.component';
-import { recipesReducer, loadedRecipesReducer } from './store/recipes.store';
+import { RecipesAndFiltersComponent } from './components/recipes-and-filters/recipes-and-filters.component';
+import { RecipesComponent } from './components/recipes/recipes.component';
+import { RecipeDetailsComponent } from './components/recipe-details/recipe-details.component';
+import { RecipeComponent } from './components/recipe/recipe.component';
+import { FiltersComponent } from './components/filters/filters.component';
+import { CardComponent } from './components/card/card.component';
 import { FeaturesComponent } from './components/features/features.component';
 
-let rootReducer = { recipes: recipesReducer, loadedRecipes: loadedRecipesReducer, };
+import { Backend } from './services/backend';
+import { MaterialIconsService } from './services/material-icons/material-icons.service';
+import { Recipe} from './models/recipe.model';
+import { appReducer } from './store/reducers/reducers';
+import { RecipesEffects } from './store/effects/effects';
+import { initialState } from './models/state.model';
 
 @NgModule({
   declarations: [
     AppComponent,
+    CardComponent,
+    FeaturesComponent,
     HeaderComponent,
     SidenavComponent,
-    FeaturesComponent,
+    RecipesAndFiltersComponent,
+    RecipesComponent,
+    RecipeDetailsComponent,
+    RecipeComponent,
+    FiltersComponent
   ],
   imports: [
     BrowserModule,
-    RouterModule.forRoot(routes),
-    SharedModule.forRoot(),
-    StoreModule.provideStore(rootReducer),
+    ReactiveFormsModule,
+    HttpModule,
+    NoopAnimationsModule,
+    MdInputModule,
+    MdCheckboxModule,
+    MdToolbarModule,
+    MdButtonModule,
+    MdSidenavModule,
+    MdIconModule,
+    MdListModule,
+    MdOptionModule,
+    MdSelectModule,
+    MdProgressSpinnerModule,
+    MdCardModule,
+    RouterModule.forRoot([
```

Comment:
maybe this could be moved to a separate routes file. The app is small now, but when they get large this can bloat the app.component file a lot.